models/classes/runner/time/QtiTimer.php

```diff
@@ -420,10 +411,22 @@ protected function unserializeTimeLine($data)
         } else {
             $timeLine = $data;
         }
-        
+
         return $timeLine;
     }
 
+    protected function unserializeAdjustmentMap($data)
+    {
+        if (is_array($data)) {
+            $map = new AdjustmentMap();
+            $map->fromArray($data);
+        } else {
```

Comment:
Maybe we have to check if `$map instanceof TimerAdjustmentMapInterface` before we assign it.

Reply:
Done